crates/rpc/rpc-builder/src/error.rs

```diff
@@ -62,3 +69,19 @@ impl RpcError {
         }
     }
 }
+
+/// Errors when trying to launch ws and http server on the same port.
+#[derive(Debug, thiserror::Error)]
+pub enum WsHttpSamePortError {
+    /// Ws and http server configured on same port but with different cors domains.
+    #[error("CORS domains for http and ws are different, but they are on the same port")]
+    ConflictingCorsDomains,
+    /// Ws and http server configured on same port but with different modules.
+    #[error("Different api modules for http and ws on the same port is currently not supported: http: {http_modules:?}, ws: {ws_modules:?}")]
+    ConflictingModules {
+        /// Http modules.
+        http_modules: Vec<RethRpcModule>,
+        /// Ws modules.
+        ws_modules: Vec<RethRpcModule>,
+    },
```

Comment:
given the jsonrpsee limitation this is a good way to explain the lack of support